typescript/packages/wallets/evm/src/EVMWalletClient.ts

```diff
@@ -1,9 +1,498 @@
+                    decimals.value === undefined ||
+                    name.value === undefined ||
+                    symbol.value === undefined
+                ) {
+                    throw new Error("Incomplete token information received.");
+                }
+
+                const balanceInBaseUnits = String(rawBalance.value);
+                const tokenDecimals = Number(decimals.value);
+                const balanceValue = formatUnits(BigInt(balanceInBaseUnits), tokenDecimals);
+
+                return {
+                    decimals: tokenDecimals,
+                    symbol: String(symbol.value),
+                    name: String(name.value),
+                    value: balanceValue,
+                    inBaseUnits: balanceInBaseUnits,
+                };
+            } catch (error) {
+                console.error(`Failed to fetch ERC20 balance for ${tokenAddress}: ${error}`);
+                throw new Error(
+                    `Failed to fetch balance for token ${tokenAddress}: ${error instanceof Error ? error.message : String(error)}`,
+                );
+            }
+        }
+
+        // Native Balance
+        try {
+            const balanceInBaseUnits = await this.getNativeBalance(); // Use the new abstract method
+            const nativeDecimals = 18; // Standard for EVM native currency
+            const balanceValue = formatUnits(balanceInBaseUnits, nativeDecimals);
+
+            return {
+                decimals: nativeDecimals,
+                symbol: chain.nativeCurrency.symbol, // Get from chain config
+                name: chain.nativeCurrency.name, // Get from chain config
+                value: balanceValue,
+                inBaseUnits: String(balanceInBaseUnits),
+            };
+        } catch (error) {
+            console.error(`Failed to fetch native balance: ${error}`);
+            throw new Error(
+                `Failed to fetch native balance: ${error instanceof Error ? error.message : String(error)}`,
+            );
+        }
+    }
+
+    async getTokenInfoByTicker(ticker: string) {
+        const chain = this.getChain() as EvmChain;
+        const upperCaseTicker = ticker.toUpperCase();
+
+        // Check configured tokens first
+        const token = this.tokens.find((t) => t.symbol.toUpperCase() === upperCaseTicker);
+
+        if (token) {
+            const contractAddress = token.chains[chain.id]?.contractAddress;
+            if (!contractAddress) {
+                throw new Error(`Token ${ticker} configured but not found on chain ${chain.id}`);
+            }
+            // Return info from configuration
+            return {
+                symbol: token.symbol,
+                contractAddress: contractAddress,
+                decimals: token.decimals,
+                name: token.name,
+            };
+        }
+
+        // Check if it's the native currency symbol
+        if (upperCaseTicker === chain.nativeCurrency.symbol.toUpperCase()) {
```

Comment:
I would also check if `symbol === ETH` here since a chain might have another name of the token but people still call it eth